snapcraft/plugins/_python/_pip.py

```diff
@@ -372,11 +373,23 @@ def list(self, *, user=False):
         if user:
             command.append('--user')
 
-        output = self._run_output(command)
         packages = collections.OrderedDict()
         try:
+            output = self._run_output(command)
             json_output = json.loads(
                     output, object_pairs_hook=collections.OrderedDict)
+        except subprocess.CalledProcessError:
+            # --format requires a newer pip, so fall back to legacy output
+            command = ['list']
+            if user:
+                command.append('--user')
```

Comment:
We're asking for trouble by duplicating this logic. Perhaps the [base command](https://github.com/snapcore/snapcraft/pull/2055/files#diff-c4fa5148d03d9fab1aa1f2284d873aa6R372) should be `[list]` with `--format=json` thrown on at the last minute in the `try`, so we can just re-use the original `command` (and its `--user` logic).

Reply:
Done.